### PR TITLE
HW-53465, HW-53468. Renamed listReceipts to listUserReceipts, tests

### PR DESCRIPTION
**Pull request by azakrevska-epam**

### CODE REVIEW

File: core/src/main/java/com/hyperwallet/android/Hyperwallet.java [hunk 1]
```diff
@@ -710,7 +710,7 @@ public void retrieveTransferMethodConfigurationFields(
      * @param receiptQueryParam the ordering and filtering criteria
      * @param listener          the callback handler of responses from the Hyperwallet platform; must not be null
      */
-    public void listReceipts(@NonNull final ReceiptQueryParam receiptQueryParam,
+    public void listUserReceipts(@NonNull final ReceiptQueryParam receiptQueryParam,
```
Comment (peter-joseph) [inline core/src/main/java/com/hyperwallet/android/Hyperwallet.java]: Second thought `ReceiptQueryParam` can be null its never required to pass in parameters please change this to `@Nullable` and then use the built in function to check if query param is valid or not `buildUrlQueryIfRequired `

File: core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletListUserReceiptsTest.java [hunk 2]
```diff
@@ -48,7 +48,7 @@
 import okhttp3.mockwebserver.RecordedRequest;
 
 @RunWith(RobolectricTestRunner.class)
-public class HyperwalletListReceiptsTest {
+public class HyperwalletListUserReceiptsTest {
```
Comment (peter-joseph) [inline core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletListUserReceiptsTest.java]: if you can select one of the test cases below that demonstrates nullable query parameter please do so
so that you can also verify url assemble without `?` end the end of the URL

**Comment by peter-joseph** [comment]
> LGTM 👍

In second thought please have a look on comments

File: core/src/main/java/com/hyperwallet/android/Hyperwallet.java (hunk 1 above)
Comment (azakrevska-epam) [reply core/src/main/java/com/hyperwallet/android/Hyperwallet.java]: @peter-joseph Done

File: core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletListUserReceiptsTest.java (hunk 2 above)
Comment (azakrevska-epam) [reply core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletListUserReceiptsTest.java]: Added

Comment (fmattos-hw) [reply core/src/test/java/com/hyperwallet/android/transfermethod/HyperwalletListUserReceiptsTest.java]: If this test is not required due to jacoco coverage, I'd rather not have this test case. Also in Hyperwallet#listUserReceipts the query param should be non null instead. Just think, we say the query param is nullable then we add a test to verify that if the query is null there's no data returned... It doesn't make much sense. We can discuss this offline but the case testListUserReceipts_withEmptyParam_returnEmptyResponse seems not necessary and the param in the list method should be non null

File: core/src/main/java/com/hyperwallet/android/Hyperwallet.java (hunk 1 above)
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/Hyperwallet.java]: hi @azakrevska-epam, please make the query param as @Nullable - After discussing with the rest of the team, we decided that nullable is the best way

Comment (azakrevska-epam) [reply core/src/main/java/com/hyperwallet/android/Hyperwallet.java]: @fmattos-hw Done, added test for checking absence of offset and limit
